framework/src/utils/LibtorchArtificialNeuralNet.C

```diff
@@ -0,0 +1,171 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
+#include "LibtorchArtificialNeuralNet.h"
+#include "MooseError.h"
+
+namespace Moose
+{
+
+LibtorchArtificialNeuralNet::LibtorchArtificialNeuralNet(
+    const std::string name,
+    const unsigned int num_inputs,
+    const unsigned int num_outputs,
+    const std::vector<unsigned int> & num_neurons_per_layer,
+    const std::vector<std::string> activation_function)
```

Comment:
I would like to have a default value for the activation functions: relu, in the H file I have 

```
LibtorchArtificialNeuralNet(const std::string name,
                              const unsigned int num_inputs,
                              const unsigned int num_outputs,
                              const std::vector<unsigned int> & num_neurons_per_layer,
                              const std::vector<std::string> activation_function = {"relu"});
```

So I'd like to keep it the way it is. This will be most likely a vector of 10-20 strings so I think copying is not the works. Also, this function is only called once at constructions so I don't expect to have a lot of overhead because of the copy argument.

Reply:
I take this comment back, looks like you can actually do this with a const reference.